catalog-rest-service/src/main/java/org/openmetadata/catalog/resources/util/UtilResource.java

```diff
@@ -0,0 +1,100 @@
+import org.openmetadata.catalog.jdbi3.CollectionDAO;
+import org.openmetadata.catalog.jdbi3.UtilRepository;
+import org.openmetadata.catalog.resources.Collection;
+import org.openmetadata.catalog.security.Authorizer;
+import org.openmetadata.catalog.util.EntitiesCount;
+import org.openmetadata.catalog.util.ServicesCount;
+
+@Path("/v1/util")
+@Api(value = "Util collection", tags = "Util collection")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+@Collection(name = "util")
+@Slf4j
+public class UtilResource {
+  public static final String COLLECTION_PATH = "/v1/util";
+  private final UtilRepository utilRepository;
+  private final Authorizer authorizer;
+
+  public UtilResource(CollectionDAO dao, Authorizer authorizer) {
+    Objects.requireNonNull(dao, "UtilRepository must not be null");
+    this.utilRepository = new UtilRepository(dao.utilDAO());
+    this.authorizer = authorizer;
+  }
+
+  @GET
+  @Path("/entities/count")
+  @Operation(
+      operationId = "listEntitiesCount",
+      summary = "List All Entities Counts",
+      tags = "util",
+      description = "Get a List of all Entities Count",
+      responses = {
+        @ApiResponse(
+            responseCode = "200",
+            description = "List of Entities Count",
+            content = @Content(mediaType = "application/json", schema = @Schema(implementation = EntitiesCount.class)))
+      })
+  public EntitiesCount listEntitiesCount(@Context UriInfo uriInfo) {
+    return utilRepository.getAllEntitiesCount();
+  }
+
+  @GET
+  @Path("/services/count")
+  @Operation(
+      operationId = "listServicesCount",
+      summary = "List All Services Counts",
+      tags = "util",
+      description = "Get a List of all Entities Count",
+      responses = {
+        @ApiResponse(
+            responseCode = "200",
+            description = "List of Services Count",
+            content = @Content(mediaType = "application/json", schema = @Schema(implementation = ServicesCount.class)))
+      })
+  public ServicesCount listServicesCount(@Context UriInfo uriInfo) {
+    return utilRepository.getAllServicesCount();
+  }
+
+  @GET
+  @Path("/{entity}/count")
+  @Operation(
+      operationId = "listEntityCount",
+      summary = "List Individual Entity Count",
+      tags = "util",
+      description = "Get total count of an Entity",
+      responses = {
+        @ApiResponse(
+            responseCode = "200",
+            description = "List of Services Count",
+            content = @Content(mediaType = "application/json"))
+      })
+  public Object listEntityCount(
+      @Context UriInfo uriInfo,
+      @Parameter(
+              description = "Entity type for which usage is requested",
+              required = true,
+              schema = @Schema(type = "string", example = "table, pipeline, topic, or dashboard"))
+          @PathParam("entity")
+          String entity) {
+    return utilRepository.getIndividualEntityCount(entity);
```

Comment:
We should avoid sending raw integer responses. Instead a JSON response like `{count: 5}` would be much better for individual entity count. One more thing to note here is that the `getIndividualEntityCount(entity)` method does not provide any performance advantage over `getAllEntitiesCount()` since it is going to hit the database with the same query to get all the counts.

Reply:
We can get rid of the getIndividualEntityCount() method since the same already exists with "limit=0" param